server/db/model.py

```diff
@@ -41,8 +41,18 @@ class Prospect(MongoModel):
     first_name = fields.CharField()
     last_name = fields.CharField()
     status = fields.CharField()
+    campaigns = fields.ListField(fields.ReferenceField(
+        "Campaign"))
+
+    @staticmethod
+    def find_by_id(id):
+        prospect = Prospect.objects.raw({'_id': id})
```

Comment:
Should change to `prospect = Prospect.objects.get({'_id': id})`; But this might raise NotExistError if item not found.

Reply:
as raw will return a queryset; you can also do `list(Prospect.objects.raw({'_id': id}))` to get a list

Reply:
Would wrapping in a try/except block handle that?

        try:
            prospect = Prospect.objects.get({'_id': id})
            return prospect
        except:
            return None

or would I need to specify which exception to check for

Reply:
> Would wrapping in a try/except block handle that?
> 
> ```
>     try:
>         prospect = Prospect.objects.get({'_id': id})
>         return prospect
>     except:
>         return None
> ```
> 
> or would I need to specify which exception to check for

You can either do that, or just let the error raise and handle the error with app_errorhandler (see last line of campaign_handler.py)